### ISSUE TITLE
sonata_type_collection - clicking add button returns response but it isn't added to the page!

### ISSUE THREAD

**Issue by Tocacar** [sonata-project/SonataAdminBundle#2702]
I have got this working fine on other admin classes, it's just this one class that this weird behaviour is happening for.  I've been through it a hundred times and can't figure it out:

```
//composer.json (the sonata bits) - ran update today
"sonata-project/cache-bundle": "2.2.*@dev",
        "sonata-project/block-bundle": "2.2.*",
        "sonata-project/admin-bundle": "2.3.*",
        "sonata-project/core-bundle": "~2.3",
        "sonata-project/exporter": "1.*",
        "sonata-project/doctrine-orm-admin-bundle": "2.3.*",
        "sonata-project/easy-extends-bundle" : "2.1.7",
        "sonata-project/user-bundle": "2.3.*@dev",
        "sonata-project/datagrid-bundle": "2.2.*@dev",
```

```
//LitterAdmin.php config

//other stuff
->add('pups', 
           'sonata_type_collection', array(
                'by_reference' => false 
            ), array(
                'edit' => 'inline',
                'inline' => 'table'
            )
);

```

 ```
//Litter.php mapping

    /**
     * @var ArrayCollection $pups
     *
     * @ORM\OneToMany(targetEntity="Mouse", mappedBy="litter", cascade={"persist"}, orphanRemoval=true)
     */
    private $pups;

/**
     * Add pup
     *
     * @param \AMS\ColonyBundle\Entity\Mouse $pup
     * @return Litter
     */
    public function addPup(\AMS\ColonyBundle\Entity\Mouse $pup)
    {
        $this->pups[] = $pup;
        $pup->setLitter($this);
    
        return $this;
    }

    /**
     * Remove pup
     *
     * @param \AMS\ColonyBundle\Entity\Mouse $pup
     */
    public function removePup(\AMS\ColonyBundle\Entity\Mouse $pup)
    {
        $this->pups->removeElement($pup);
    }

    /**
     * Get pups
     *
     * @return \Doctrine\Common\Collections\Collection 
     */
    public function getPups()
    {
        return $this->pups;
    }
```

```
//Mouse.php mapping

    /**
     * @var integer $litter_id
     *
     * @ORM\Column(name="litter_id", type="integer", nullable=true)
     */
    private $litter_id;

    /**
     * @ORM\ManyToOne(targetEntity="Litter", inversedBy="pups")
     * @ORM\JoinColumn(name="litter_id", referencedColumnName="id")
     */
    protected $litter;

    /**
     * Set litter
     *
     * @param \AMS\ColonyBundle\Entity\Litter $litter
     * @return Mouse
     */
    public function setLitter(\AMS\ColonyBundle\Entity\Litter $litter = null)
    {
        $this->litter = $litter;
    
        return $this;
    }

    /**
     * Get litter
     *
     * @return \AMS\ColonyBundle\Entity\Litter 
     */
    public function getLitter()
    {
        return $this->litter;
    }

```

Now... in the LitterAdmin form, when I press the 'Add' button, this is the content of the POST response from Firebug:

```
              
    <div id="field_container_s54ce677183ded_pups" class="field-container">
        <span id="field_widget_s54ce677183ded_pups" >
                                                                        <table class="table table-bordered">
                            <thead>
                                <tr>
                                                                    </tr>
                            </thead>
                            <tbody class="sonata-ba-tbody">
                                                                    <tr>
                                                                            </tr>
                                                            </tbody>
                        </table>
                                                
        </span>

        
                            <span id="field_actions_s54ce677183ded_pups" >
                    <a
                        href="/admin/ams/colony/mouse/create?uniqid=s54ce6776da8f9&amp;code=ams.admin.mouse&amp;pcode=ams.admin.litter&amp;puniqid=s54ce677183ded"
                        onclick="return start_field_retrieve_s54ce677183ded_pups(this);"
                        class="btn btn-success btn-sm btn-outline sonata-ba-action"
                        title="Add new"
                        >
                        <i class="fa fa-plus-circle"></i>
                        Add new
                    </a>
                </span>
            
                        
                        



<!-- edit one association -->

<script type="text/javascript">

    // handle the add link
    var field_add_s54ce677183ded_pups = function(event) {

        event.preventDefault();
        event.stopPropagation();

        var form = jQuery(this).closest('form');

        // the ajax post
        jQuery(form).ajaxSubmit({
            url: 'http://ams.ldev/admin/core/append-form-field-element?code=ams.admin.litter&elementId=s54ce677183ded_pups&uniqid=s54ce677183ded',
            type: "POST",
            dataType: 'html',
            data: { _xml_http_request: true },
            success: function(html) {
                if (!html.length) {
                    return;
                }

                jQuery('#field_container_s54ce677183ded_pups').replaceWith(html); // replace the html

                Admin.shared_setup(jQuery('#field_container_s54ce677183ded_pups'));

                if(jQuery('input[type="file"]', form).length > 0) {
                    jQuery(form).attr('enctype', 'multipart/form-data');
                    jQuery(form).attr('encoding', 'multipart/form-data');
                }
                jQuery('#sonata-ba-field-container-s54ce677183ded_pups').trigger('sonata.add_element');
                jQuery('#field_container_s54ce677183ded_pups').trigger('sonata.add_element');
            }
        });

        return false;
    };

    var field_widget_s54ce677183ded_pups = false;

    // this function initialize the popup
    // this can be only done this way has popup can be cascaded
    function start_field_retrieve_s54ce677183ded_pups(link) {

        link.onclick = null;

        // initialize component
        field_widget_s54ce677183ded_pups = jQuery("#field_widget_s54ce677183ded_pups");

        // add the jQuery event to the a element
        jQuery(link)
            .click(field_add_s54ce677183ded_pups)
            .trigger('click')
        ;

        return false;
    }
</script>

<!-- / edit one association -->


            </div>
```

But nothing appears on the page!  No Firebug errors, and Symfony 2.6.3 debug bar doesn't register any AJAX action either.

Can anyone see what's wrong with this?

**Comment by Tocacar** [comment]
And I'm not overriding any Sonata templates at all.

**Comment by Tocacar** [comment]
...and here is my sonata admin services config:

```
    ams.admin.mouse:
        class:  AMS\AdminBundle\Admin\MouseAdmin
        tags:
          - { name: sonata.admin, manager_type: orm, group: ams_colony, label: Mice }
        arguments: [null, AMS\ColonyBundle\Entity\Mouse, SonataAdminBundle:CRUD]
        calls:
          - [ addChild, [ @ams.admin.occupancy] ]
          - [ addChild, [ @ams.admin.mouse.comment] ]

    ams.admin.litter:
        class:  AMS\AdminBundle\Admin\LitterAdmin
        tags:
          - { name: sonata.admin, manager_type: orm, group: ams_colony, label: Litter }
        arguments: [null, AMS\ColonyBundle\Entity\Litter, SonataAdminBundle:CRUD]
```

**Comment by Tocacar** [comment]
I just re-tested one of the admin classes where sonata_type_collection IS working and verified that sf2.6.3's debug toolbar doesn't register any AJAX action there either, so I guess that bit of the puzzle is not relevant.

**Comment by Tocacar** [comment]
Solved it.  My Mouse entity has a parent/children self-referencing set-up going on, and I had 

```
//MouseAdmin.php

->add('children')
```

...in my MouseAdmin.php class.  As soon as I commented that out, the embedded MouseAdmin form was added to the LitterAdmin form page.

Presumably this is just a limitation of what's possible with embedded forms.  I'll just conditionally remove this form field in the MouseAdmin class to work around it.
